cdn_lambda/functions/origin_response/origin_response.py

```diff
@@ -24,6 +25,22 @@ def handler(self, event, context):
         request = event["Records"][0]["cf"]["request"]
         response = event["Records"][0]["cf"]["response"]
 
+        try:
+            want_digest = request["headers"]["want-digest"][0]["value"]
+        except KeyError:
+            LOG.debug(
+                "Could not read want-digest from response", exc_info=True
+            )
+            want_digest = None
+
+        if want_digest == "sha-256":
+            response["headers"]["digest"] = [
```

Comment:
Question: do you also have to check whether the response from origin was successful?  If we got an error or other unusual response, I don't think we should add the digest.  This is a question since I don't know whether the origin-response event is designed so that the lambda is invoked also on failures (I guess that it probably is, since there's valid use-cases to do things with failures from origin).

Reply:
I'm only able to get any of the custom response headers, including this one, to appear on successful requests;
`HTTP/2 200 
content-type: binary/octet-stream
content-length: 2475
date: Mon, 30 Mar 2020 20:20:12 GMT
last-modified: Wed, 19 Feb 2020 22:29:14 GMT
etag: "038ce56c7a07d5f070572f1bf19fb939"
accept-ranges: bytes
server: AmazonS3
digest: sha-256=f8fc113a93a12fe6e2713cf550cd7d015d8defbdcb6fa00f9b85863b18c3efe4
x-cache: Miss from cloudfront
via: 1.1 11e42b3facdb2abb175876ba0e4b97f9.cloudfront.net (CloudFront)
x-amz-cf-pop: IAD79-C3
x-amz-cf-id: 922kbbsaKhGlQheZrlse-mnqDxwtdage4W3b6VfGieL2LokF65Fj3Q==`

Creating a table item which points to an invalid object key got me the following;

`HTTP/2 404 
content-length: 0
server: CloudFront
date: Mon, 30 Mar 2020 19:54:24 GMT
x-cache: Error from cloudfront
via: 1.1 15cde442051269a0307a638d23683c8c.cloudfront.net (CloudFront)
x-amz-cf-pop: IAD79-C3
x-amz-cf-id: nNilA3dXMxW07j2nL6IX7rpNUsaD9rsOQoCiwawNEMPaD3kLr46UkA==
age: 185`

If we want to be sure, we could check content-length header.